include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -1002,11 +1002,18 @@ namespace LinearAlgebra
     void
     Vector<Number, MemorySpaceType>::zero_out_ghost_values() const
     {
-      Kokkos::pair<size_type, size_type> range(
-        partitioner->locally_owned_size(),
-        partitioner->locally_owned_size() + partitioner->n_ghost_indices());
-      if (data.values.size() > 0)
-        Kokkos::deep_copy(Kokkos::subview(data.values, range), 0);
+      if (std::is_same_v<MemorySpaceType, MemorySpace::Host>)
+        std::memset(data.values.data() + partitioner->locally_owned_size(),
+                    0,
+                    partitioner->n_ghost_indices() * sizeof(Number));
```

Comment:
Would you mind adding a comment here describing that we are rather using `memset` heredircetly for performance/overhead reasons?

Reply:
Good idea, I did that.